config/notifiers.go

```diff
@@ -579,3 +589,46 @@ func (c *PushoverConfig) UnmarshalYAML(unmarshal func(interface{}) error) error
 	}
 	return nil
 }
+
+// TODO: Move to common?
+
+// SigV4Config is the configuration for signing remote write requests with
+// AWS's SigV4 verification process. Empty values will be retrieved using the
+// AWS default credentials chain.
+type SigV4Config struct {
+	Region    string `yaml:"region,omitempty"`
+	AccessKey string `yaml:"access_key,omitempty"`
+	SecretKey Secret `yaml:"secret_key,omitempty"`
+	Profile   string `yaml:"profile,omitempty"`
+	RoleARN   string `yaml:"role_arn,omitempty"`
+}
+
+type SNSConfig struct {
+	NotifierConfig `yaml:",inline" json:",inline"`
+
+	HTTPConfig *commoncfg.HTTPClientConfig `yaml:"http_config,omitempty" json:"http_config,omitempty"`
+
+	APIUrl      string            `yaml:"api_url" json:"api_url"`
+	APIVersion  string            `yaml:"api_version,omitempty" json:"api_version,omitempty"`
+	Sigv4       SigV4Config       `yaml:"sigv4" json:"sigv4"`
+	TopicARN    string            `yaml:"topic_arn,omitempty" json:"topic_arn,omitempty"`
+	PhoneNumber string            `yaml:"phone_number,omitempty" json:"phone_number,omitempty"`
+	TargetARN   string            `yaml:"target_arn,omitempty" json:"target_arn,omitempty"`
+	IsFIFOTopic bool              `yaml:"is_fifo_topic,omitempty" json:"is_fifo_topic,omitempty"`
```

Comment:
Looks a weird naming to me. Also I believe it's not much future proof, cause if a 3rd SNS queue type will be added it will not fit very well into a boolean. What's about a `queue_type` whose values can be either `standard` (default) or `fifo`?

Reply:
Another option is to use the topic/target ARN to decide if the topic is FIFO or Standard. ARN's for topics that are FIFO will end in `.fifo`. Using the suffix to determine queue type is noted here in the aws docs https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/FIFO-queues.html. I think using the topic/target arn would be the right path forward here rather than the config option.

Reply:
What is the FIFO info used for?
Shouldn't the topic be created by the tenant and used by the receiver independently whether it is FIFO or not??

Reply:
The FIFO information is used to decide if we need to send a FIFO required de-duplication key and message group id as part of the SNS publish action. But we can figure that out from the topic ARN.

Reply:
> The FIFO information is used to decide if we need to send a FIFO required de-duplication key and message group id as part of the SNS publish action

What happens if we set it for standard queues too? Does the Publish API call fails?

Reply:
Yes, if we send a de-duplication key to a standard type queue the publish fails with: `Invalid parameter: MessageDeduplicationId Reason: The request includes MessageDeduplicationId parameter that is not valid for this topic type`

Reply:
Is using `GetTopicAttributes()` an option (and the cache it in-memory on a per ARN basis)? I think would be safer than parsing the ARN.

Reply:
I switched the check to use `GetTopicAttributes()` but I couldn't find a good place to cache the result, any suggestions on where to cache Topics to isFifo booleans?

Reply:
Discussed offline. We made it work. Will be fixed in the next commit.

Reply:
Had some chat with @treid314 offline. Here is my take on calling `GetTopicAttributes()` to tell if a topic is FIFO. 

While it works, but it has the drawback that:
1. it's extra API call
2. Customer has to explicitly give sender permission to call `GetTopicAttributes()` on top of `Publish()`, the `GetTopicAttribute()` returns a lot of detailed information on the topic (e.g. IAM policy, KMS Key ID) that customer may not want to expose to the sender. 

The `.fifo` suffix was employed by SNS for senders to easily tell the topic type without making extra API call. So I would suggest we just use the suffix to tell the topic type.